### PR TITLE
compat api: allow default bridge name for networks

### LINKED ISSUE #14983: Cannot create network with name "bridge" because it conflicts with a valid network mode
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

Cannot create a network named bridge to make badly written applications (but not changeable) that depend on Docker default network name work.

**Steps to reproduce the issue:**

1. podman network create bridge

**Describe the results you received:**

Error: cannot create network with name "bridge" because it conflicts with a valid network mode

**Describe the results you expected:**

Network is created.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
4.1.0
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
(42/43) Installing podman (4.1.0-r1)
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/main/troubleshooting.md)**

Yes

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Podman inside Alpine container.

### PR DESCRIPTION
**Pull request by Luap99**
Docker uses "bridge" as default network name so some tools expect this
to work with network list or inspect. To fix this we change "bridge" to
the podman default ("podman") name.

Fixes #14983

#### Does this PR introduce a user-facing change?

```release-note
The network endpoints for the docker compat API will now correctly use "bridge" as default network name.
Networks created via the docker compat API are now always isolated from each other (e.g. network1 cannot connect to network2) to match the docker behaviour. 
```

### CODE REVIEW

**Comment by Luap99** [comment]
@mheon PTAL

**Comment by rhatdan** [comment]
Validate does not like your whitespace.

File: pkg/api/handlers/compat/networks.go [hunk 1]
```diff
@@ -107,11 +114,16 @@ func convertLibpodNetworktoDockerNetwork(runtime *libpod.Runtime, network nettyp
 		Config:  ipamConfigs,
 	}
 
+	name := network.Name
+	if changeDefaultName && name == runtime.Network().DefaultNetworkName() {
+		name = nettypes.BridgeNetworkDriver
+
```
Comment (mheon) [inline pkg/api/handlers/compat/networks.go:120]: Extra whitespace?

File: pkg/api/handlers/compat/networks.go [hunk 2]
```diff
@@ -182,27 +194,20 @@ func CreateNetwork(w http.ResponseWriter, r *http.Request) {
...
 	for opt, optVal := range networkCreate.Options {
 		switch opt {
-		case "mtu":
+		case nettypes.MTUOption:
 			fallthrough
 		case "com.docker.network.driver.mtu":
-			if network.Driver == nettypes.BridgeNetworkDriver {
-				network.Options["mtu"] = optVal
-			}
-		case "icc":
```
Comment (mheon) [inline pkg/api/handlers/compat/networks.go:194]: Why was this removed?

**Comment by mheon** [comment]
Left some comments. For the overall approach, I honestly hate it but I can't think of a better way. LGTM once comments addressed.

File: pkg/api/handlers/compat/networks.go (hunk 2 above)
Comment (Luap99) [reply pkg/api/handlers/compat/networks.go:194]: see commit message icc is not supported by us, it can just fall into the default case.

**Comment by Luap99** [comment]
> Left some comments. For the overall approach, I honestly hate it but I can't think of a better way. LGTM once comments addressed.

I don't like it either, the only other way would be to fix this in the network backend then all podman APIs/commands could use it as well. But this would mean to fix/maintain it for both the cni and netavark backend.
